pom.xml

```diff
@@ -78,11 +78,27 @@
         </dependency>
 
         <!-- test dependencies -->
+
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+            <!-- Todo: remove explicit version when 3rd party BOM 3.22.2 is released-->
+            <version>5.8.1</version>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
```

Comment:
We could remove the old JUnit4 dependency

Reply:
This is just the junit 5 vintage one. I only put it in there to keep the "peripheral" changes to a minimum in this PR, it would be trivial to move to native JUnit 5 as there's only 40-something tests

Reply:
Yepp. But I was referring to the plain old JUnit4 dependency and not the vintage.